scalding-core/src/main/scala/com/twitter/scalding/Job.scala

```diff
@@ -116,6 +116,9 @@ class Job(val args: Args) extends FieldConversions with java.io.Serializable {
   implicit def iterableToRichPipe[T](iter: Iterable[T])(implicit set: TupleSetter[T], conv: TupleConverter[T]): RichPipe =
     RichPipe(toPipe(iter)(set, conv))
 
+  // Provide args as an implicit val for extensions such as the Checkpoint extension.
+  implicit protected val _args: Args = args
```

Comment:
1) let's make this a def (avoid having two fields with the same data).
2) to decrease the chances with collide with another name can we give it a longer name? `_implicitJobArgs` or something?